### PR TITLE
Add separate test to ensure that `delegate` with `:private` option returns correct value

### PR DESCRIPTION
**Pull request by bogdanvlviv**
<s>Remove extra comments `# Asking for private method` in activesupport/test/core_ext/module_test.rb</s>
Improve docs of using `delegate` with `:private`
Update changelog of #31944

### CODE REVIEW

File: activesupport/test/core_ext/module_test.rb [hunk 1]
```diff
@@ -417,7 +417,7 @@ def initialize(place)
     assert_not_respond_to place, :street
     assert_not_respond_to place, :city
 
-    assert place.respond_to?(:street, true) # Asking for private method
```
Comment (kamipo) [inline activesupport/test/core_ext/module_test.rb]: This is the reason why we can not use `assert_respond_to`. Looks not a noise for me.

Comment (rafaelfranca) [reply activesupport/test/core_ext/module_test.rb]: Yup. Agree with @kamipo

Comment (bogdanvlviv) [reply activesupport/test/core_ext/module_test.rb]: We use `respond_to?` for the same reason in 3 more tests(in `activesupport/test/core_ext/module_test.rb`). Do you think we should explain it in each of them?

Comment (bogdanvlviv) [reply activesupport/test/core_ext/module_test.rb]: I agree, I'll revert this cosmetic changes and won't add additional ones then

**Pull request merged**